tasks/api/src/main/java/org/projectnessie/nessie/tasks/api/TaskState.java

```diff
@@ -69,27 +70,52 @@ public interface TaskState {
   @JsonInclude(JsonInclude.Include.NON_NULL)
   String message();
 
-  TaskState SUCCESS = ImmutableTaskState.of(TaskStatus.SUCCESS, null, null, null);
+  /**
+   * Represents the deadline when a task will fail even if it gets a re-triable error. Note: passing
+   * the deadline may or may not forcibly cancel a running task.
+   */
+  @Value.Parameter(order = 5)
+  @Nullable
+  @JsonInclude(JsonInclude.Include.NON_NULL)
+  @JsonDeserialize(using = InstantAsLongDeserializer.class)
+  @JsonSerialize(using = InstantAsLongSerializer.class)
+  Instant deadline();
+
+  TaskState SUCCESS = ImmutableTaskState.builder().status(TaskStatus.SUCCESS).build();
```

Comment:
Interesting that this is not marked as a potential class-loader deadlock

Reply:
This is an `interface`, I suppose the JVM does not need to resolve all its static fields before it can use it.... In a way static fields on an `interfaces` are just global constants.

Reply:
I'm surprised, because there should actually no real difference.

Reply:
I can still move it to a separate class if you prefer... WDYT?

Reply:
Seems to be fine